lua/starfall/libs_sh/vectors.lua

```diff
@@ -189,10 +189,17 @@ function vec_meta.__eq(a, b)
 	return a[1]==b[1] and a[2]==b[2] and a[3]==b[3]
 end
 
+local math_asin, math_atan2, math_sqrt = math.asin, math.atan2, math.sqrt
+local rad2deg = 180 / math.pi
+
 --- Get the vector's angle.
 -- @return Angle Angle representing the vector
 function vec_methods:getAngle()
-	return awrap(unwrap(self):Angle())
+	return setmetatable({
```

Comment:
I think the `%360` can be removed. Also if the vector length `math_sqrt(self[1]^2 + self[2]^2 + self[3]^2)` is zero, we should return a zero angle. Currently, it returns an angle with nan pitch.

Can you do the performance test again after that change?

Reply:
the % 360 is for parity with the original since the original never returns an angle with any negatives